src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_universal.cpp

```diff
@@ -0,0 +1,1087 @@
+        std::string(" -DCK_PARAM_DEPENDENT_GEMM_B_BLOCK_COPY_DST_DATA_PER_WRITE_GEMM_KPACK=") + std::to_string(GemmBBlockCopyDstDataPerWrite_GemmKPack) +
+        std::string(" -DCK_EXTRA_M=") + std::to_string(gemm_m_extra) +
+        std::string(" -DCK_EXTRA_N=") + std::to_string(gemm_n_extra) +
+        std::string(" -DCK_EXTRA_K=") + std::to_string(gemm_k_extra) +
+        std::string(" -DCK_USE_AMD_XDLOPS=") + std::to_string(IsXdlopsSupport(ctx) ? 1 : 0) +
+        std::string(" -DCK_USE_AMD_XDLOPS_INLINE_ASM=") + std::to_string(miopen::IsEnabled(MIOPEN_DEBUG_IMPLICIT_GEMM_XDLOPS_INLINE_ASM{}) ? 1 : 0) +
+        std::string(" -DCK_USE_AMD_XDLOPS_EMULATE=") + (miopen::IsEnabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_XDLOPS_EMULATE{}) ? '1' : '0') +
+        std::string(" -DCK_BLOCK_SYNC_LDS_WITHOUT_SYNC_VMEM=") + (miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_BLOCK_SYNC_LDS_WITHOUT_SYNC_VMEM{}) ? '0' : '1') +
+        std::string(" -DCK_WORKAROUND_SWDEV_229564=") + std::to_string(WORKAROUND_SWDEV_229564) +
+        std::string(" -DCK_WORKAROUND_SWDEV_231101=") + std::to_string(WORKAROUND_SWDEV_231101) +
+        ctx.general_compile_options;
+    // clang-format on
+
+    result.invoker_factory = conv::MakeImplGemmDataInvokerFactory(ctx);
+    result.construction_params.push_back(construction_parameters);
+    return result;
+}
+
+int ConvHipImplicitGemmForwardV4R4Xdlops_Universal::RunAndMeasureSolution(
+    const miopen::Handle& profile_h,
+    ConstData_t bot_buf,
+    Data_t top_buf,
+    ConstData_t wei_buf,
+    ConstData_t bias_buf,
+    const ConvolutionContext& ctx,
+    const ConvSolution& solution,
+    float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+bool ConvHipImplicitGemmForwardV4R4Xdlops_Universal::IsApplicable(
+    const ConvolutionContext& ctx) const
+{
+    if(ctx.skip_solutions_that_take_long_time_to_build_and_have_narrow_coverage)
+        return false;
+
+    if(!IsXdlopsSupport(ctx))
+        return false;
+
+    if(!(ctx.IsFp32() || ctx.IsFp16() || ctx.IsBfp16()))
+        return false;
+
+    if(!ctx.direction.IsForward())
+        return false;
+
+    if(!ctx.Is2d())
+        return false;
+
+    if(!IsIndexRangeLargeEnough(ctx))
+        return false;
+
+    // gemm size
+    {
+        int gemm_g       = -1;
+        int gemm_m       = -1;
+        int gemm_n       = -1;
+        int gemm_k_total = -1;
+        int gemm_extra_m = -1;
+        int gemm_extra_n = -1;
+        int gemm_extra_k = -1;
+
+        std::tie(gemm_g, gemm_m, gemm_n, gemm_k_total, gemm_extra_m, gemm_extra_n, gemm_extra_k) =
+            CalculateGemmSize(ctx, 16, 16, 4);
+        // if you do not need padding ,skip this solver
+        if(gemm_extra_m == 0 && gemm_extra_n == 0 && gemm_extra_k == 0)
```

Comment:
This is good, we need skip this solver when `extra_m/n/k` is 0.
However there is a case, for example. If you request `m_per_block x n_per_block` is `128 x 128`, but from like NxHoxWo (gemm_n for fwd) you only can have this value to be 64x, result in `128 x 128` and pad `extra_n = 64`, which is indeed can use `128 x 64` size directly without extra value. Is there a sw logic to skip this case?

Reply:
just skip it,
we don't need to add padding to every configs or tuning time is too long...